cmd/agent.go

```diff
@@ -36,11 +36,31 @@ three agents into the distribution:
 	  throught the  gossip network
 	* Publisher agent: publish snapshots to the snapshot store`,
 	TraverseChildren: true,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		// URL parsing
+		var err error
+		advertiseAddress, _ := cmd.Flags().GetString("advertise-addr")
```

Comment:
I'll sort retrieving flag with error management. 
and not returning the first at the end.